packages/flutter_tools/bin/xcode_backend.sh

```diff
@@ -306,6 +320,36 @@ EmbedFlutterFrameworks() {
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/App.framework/App"
     RunCommand codesign --force --verbose --sign "${EXPANDED_CODE_SIGN_IDENTITY}" -- "${xcode_frameworks_dir}/Flutter.framework/Flutter"
   fi
+
+  AddObservatoryBonjourService
+}
+
+# Add the observatory publisher Bonjour service to the produced app bundle Info.plist.
+AddObservatoryBonjourService() {
+  local build_mode="$(ParseFlutterBuildMode)"
+  # Debug and profile only.
+  if [[ "${build_mode}" == "release" ]]; then
+    return
+  fi
+  local built_products_plist="${BUILT_PRODUCTS_DIR}/${INFOPLIST_PATH}"
+
+  if [[ ! -f "${built_products_plist}" ]]; then
+    EchoError "error: ${INFOPLIST_PATH} does not exist. The Flutter \"Thin Binary\" build phase must run after \"Copy Bundle Resources\"."
+    exit -1
+  fi
```

Comment:
This check is new and was not in https://github.com/flutter/flutter/pull/64988.  If the user rearranged their build scripts manually, error and tell them how to fix it.
<img width="483" alt="Screen Shot 2020-09-02 at 5 16 38 PM" src="https://user-images.githubusercontent.com/682784/92049734-15580000-ed40-11ea-9300-1e1440836a82.png">

Reply:
Well I get this error but the build script "Thin Binary" is after "Copy Bundle Resources"
<img width="213" alt="Bildschirmfoto 2020-09-09 um 11 29 50" src="https://user-images.githubusercontent.com/6883178/92581291-c7874e80-f28f-11ea-885b-66ce9cf3bfb1.png">

Reply:
@bennibau can you file a new issue and @ mention me?  I want to see what's going on in your case.

Please include the output of `flutter doctor -v` and the full exported Xcode run report.
View > Navigator > Reports > Export... 
<img width="350" alt="Screen Shot 2020-09-10 at 1 41 57 PM" src="https://user-images.githubusercontent.com/682784/92802492-66cd4400-f36b-11ea-8715-89cd92dc6df3.png">

Also, a screenshot with those build phases expanded.

Reply:
I'm also hitting this error with this build script: 

![Screen Shot 2020-10-04 at 4 42 50 PM](https://user-images.githubusercontent.com/44235716/95027760-a5030e00-0660-11eb-83a1-06a29c32887c.png)

flutter doctor -v: 

[✓] Flutter (Channel master, 1.23.0-8.0.pre.131, on Mac OS X 10.15.6 19G2021 x86_64, locale en-US)
    • Flutter version 1.23.0-8.0.pre.131 at /Users/samparks/dev/sdks/flutter
    • Framework revision 8485580bb9 (3 hours ago), 2020-10-04 14:47:02 -0400
    • Engine revision a8d7d97bef
    • Dart version 2.11.0 (build 2.11.0-185.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/samparks/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

 
[✓] Connected device (3 available)            
    • iPhone (mobile)  • 00008030-000608880E9B802E • ios            • iOS 13.7
    • Web Server (web) • web-server                • web-javascript • Flutter Tools
    • Chrome (web)     • chrome                    • web-javascript • Google Chrome 85.0.4183.121

• No issues found!